File: src/google_work_agent/api/schemas/runtime_summaries/update_runtime_mode.py
"""Runtime requested-mode command wire contract."""

from typing import Literal

from google_work_agent.api.schemas.model import ApiModel
from google_work_agent.api.schemas.runtime_summaries.get_runtime_summary import RuntimeModeStatusV1


class UpdateRuntimeModeRequest(ApiModel):
    schema_version: Literal[1]
    command_id: str
    requested_mode: Literal["AUTO", "LOCAL_GPU", "API_LLM"]


__all__ = ["RuntimeModeStatusV1", "UpdateRuntimeModeRequest"]

File: frontend/src/features/settings/api/update_runtime_mode.ts
import { requestJson } from "../../../api/client";

export type RuntimeMode = "AUTO" | "LOCAL_GPU" | "API_LLM";
export type SelectableRuntimeMode = Exclude<RuntimeMode, "AUTO">;

export function updateRuntimeMode(commandId: string, requestedMode: SelectableRuntimeMode): Promise<{ schema_version: 1; requested_mode: SelectableRuntimeMode; actual_runtime: "LOCAL_GPU" | "API_LLM" | null; fallback_reason: string | null }> {
  return requestJson("/api/v1/runtime/mode", { method: "POST", body: { schema_version: 1, command_id: commandId, requested_mode: requestedMode } });
}
